### DISCUSSION TITLE
2 dimensional array

### DISCUSSION THREAD

**Discussion by nonnatee** [NationalSecurityAgency/ghidra#5263] [Q&A] [answered]
How to use CreateArrayCmd() to create short[x][y]?

**Comment by dragonmacher** [comment, marked as answer]
You can add this to a script.  Run this at an address in your program without existing code units.

```
        int x = 2;
        int y = 3;
	ArrayDataType a1 = new ArrayDataType(new ByteDataType(), y, 1);
	CreateArrayCmd cmd = new CreateArrayCmd(currentAddress, x, a1, a1.getLength());
	boolean success = runCommand(cmd);
	if (!success) {
		println("Failed: " + cmd.getStatusMsg());
	}
```

You can create DataType objects at will in your code.  The `CreateArrayCmd` also creates an  ArrayDataType *and* applies the new type to an address in your program.

**Reply by nonnatee** [reply]
Thank you. I wasted days on ByteArrayArray() and totally forgotten that in JAVA an 2d Array is just array of byteArray  XD.
